test/validation/api/scheduler/scheduler.c

```diff
@@ -507,6 +508,120 @@ static void scheduler_test_queue_size(void)
 	CU_ASSERT_FATAL(odp_pool_destroy(pool) == 0);
 }
 
+static void scheduler_test_order_ignore(void)
+{
+	odp_queue_capability_t queue_capa;
+	odp_schedule_config_t default_config;
+	odp_pool_t pool;
+	odp_pool_param_t pool_param;
+	odp_queue_param_t queue_param;
+	odp_queue_t ordered, plain, from;
+	odp_event_t ev;
+	odp_buffer_t buf;
+	uint32_t j, queue_size, num;
+	int ret;
+
+	odp_schedule_config_init(&default_config);
+	CU_ASSERT_FATAL(odp_queue_capability(&queue_capa) == 0);
+
+	queue_size = DEFAULT_NUM_EV;
+	if (default_config.queue_size &&
+	    queue_size > default_config.queue_size)
+		queue_size = default_config.queue_size;
+
+	if (queue_capa.plain.max_size &&
+	    queue_size > queue_capa.plain.max_size)
+		queue_size = queue_capa.plain.max_size;
+
+	odp_pool_param_init(&pool_param);
+	pool_param.buf.size  = 100;
+	pool_param.buf.align = 0;
+	pool_param.buf.num   = DEFAULT_NUM_EV;
+	pool_param.type      = ODP_POOL_BUFFER;
+
+	pool = odp_pool_create("test_order_ignore", &pool_param);
+
+	CU_ASSERT_FATAL(pool != ODP_POOL_INVALID);
+
+	/* Ensure that scheduler is empty */
+	for (j = 0; j < 10;) {
+		ev = odp_schedule(NULL, ODP_SCHED_NO_WAIT);
+		CU_ASSERT(ev == ODP_EVENT_INVALID);
+
+		if (ev != ODP_EVENT_INVALID)
+			odp_event_free(ev);
+		else
+			j++;
+	}
+
+	odp_queue_param_init(&queue_param);
+	queue_param.type = ODP_QUEUE_TYPE_SCHED;
+	queue_param.sched.prio  = odp_schedule_default_prio();
+	queue_param.sched.sync  = ODP_SCHED_SYNC_ORDERED;
+	queue_param.sched.group = ODP_SCHED_GROUP_ALL;
+
+	ordered = odp_queue_create("ordered", &queue_param);
+	CU_ASSERT_FATAL(ordered != ODP_QUEUE_INVALID);
+
+	odp_queue_param_init(&queue_param);
+	queue_param.type  = ODP_QUEUE_TYPE_PLAIN;
+	queue_param.order = ODP_QUEUE_ORDER_IGNORE;
+
+	plain = odp_queue_create("plain", &queue_param);
+	CU_ASSERT_FATAL(plain != ODP_QUEUE_INVALID);
+
+	num = 0;
+	for (j = 0; j < queue_size; j++) {
+		buf = odp_buffer_alloc(pool);
+		CU_ASSERT_FATAL(buf != ODP_BUFFER_INVALID);
+
+		ev = odp_buffer_to_event(buf);
+		ret = odp_queue_enq(ordered, ev);
+
+		if (ret)
+			odp_event_free(ev);
+		else
+			num++;
+	}
+
+	CU_ASSERT(num == queue_size);
+
+	num = 0;
+	for (j = 0; j < 100 * DEFAULT_NUM_EV; j++) {
+		ev = odp_schedule(&from, ODP_SCHED_NO_WAIT);
+
+		if (ev == ODP_EVENT_INVALID)
+			continue;
+
+		CU_ASSERT(from == ordered);
+		ret = odp_queue_enq(plain, ev);
+
+		if (ret)
+			odp_event_free(ev);
+		else
+			num++;
+	}
+
+	CU_ASSERT(num == queue_size);
+
+	num = 0;
+	for (j = 0; j < 100 * DEFAULT_NUM_EV; j++) {
+		ev = odp_queue_deq(plain);
+
+		if (ev == ODP_EVENT_INVALID)
+			continue;
+
+		odp_event_free(ev);
+		num++;
+	}
+
+	CU_ASSERT(num == queue_size);
```

Comment:
There seems to be no observable behavior that would test platform's conformity to `ODP_QUEUE_ORDER_IGNORE`. In a single threaded test it is even impossible to get re-ordered events. Switching to multi-threaded scenario would also not be a solution (since re-ordering is a race-condition).
Therefore I'm not sure what does this test bring to the validation suite?

Reply:
As commit log says, it's a simple test to ensure that we at least use the option in our test suite.